File: contest 9/C.java
import java.util.*;

public class C {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] a = new int[1 << 20];
        final int ADDER = 500_000;
        MySegmentTree st = new MySegmentTree(a);

        while (n-- > 0) {
            String q = sc.next();
            int l = sc.nextInt() + ADDER;
            int r = sc.nextInt() + l - 1;
            if (q.equals("W")) {
                st.rangeSet(l, r, 0);
                int[] result = st.query(0, a.length);
                System.out.println(result[0] + " " + result[1]);
            } else {
                st.rangeSet(l, r, 1);
                int[] result = st.query(0, a.length);
                System.out.println(result[0] + " " + result[1]);
            }
        }
    }
}

class MySegmentTree {
    private int n;
    private final int[][] st;
    private final int[] setTag;
    private final int[] neutral = {0, 0, 0, 0};

    public MySegmentTree(int[] a) {
        this.n = a.length;
        int size = 2 * this.n;
        this.st = new int[size][];
        Arrays.fill(st, neutral);
        this.setTag = new int[size];
        Arrays.fill(setTag, -1);
    }

    private int[] operation(int[] x, int[] y) {
        var segments = x[3] + y[2] == 2 ? x[0] + y[0] - 1 : x[0] + y[0];
        return new int[]{segments, x[1] + y[1], x[2], y[3]};
    }

    private void push(int node, int l, int r) {
        if (setTag[node] != -1) {
            int v = setTag[node];
            if (v == 1) {
                st[node] = new int[]{1, r - l + 1, 1, 1};
            } else {
                st[node] = new int[]{0, 0, 0, 0};
            }
            if (l != r) {
                setTag[node * 2] = v;
                setTag[node * 2 + 1] = v;
            }
            setTag[node] = -1;
        }
    }

    public void rangeSet(int ql, int qr, int v) {
        rangeSet(1, 0, n - 1, ql, qr, v);
    }

    private void rangeSet(int node, int l, int r, int ql, int qr, int v) {
        push(node, l, r);
        if (qr < l || r < ql) {
            return;
        }
        if (ql <= l && r <= qr) {
            setTag[node] = v;
            push(node, l, r);
        } else {
            int m = (l + r) / 2;
            rangeSet(node * 2, l, m, ql, qr, v);
            rangeSet(node * 2 + 1, m + 1, r, ql, qr, v);
            st[node] = operation(st[node * 2], st[node * 2 + 1]);
        }
    }

    public int[] query(int ql, int qr) {
        return query(1, 0, n - 1, ql, qr);
    }

    private int[] query(int node, int l, int r, int ql, int qr) {
        push(node, l, r);
        if (qr < l || r < ql) {
            return neutral;
        }
        if (ql <= l && r <= qr) {
            return st[node];
        }
        int m = (l + r) / 2;
        return operation(query(node * 2, l, m, ql, qr), query(node * 2 + 1, m + 1, r, ql, qr));
    }
}

File: contest 9/C.py
class MySegmentTree:
    def __init__(self, a):
        self.n = len(a)
        self.a = a
        size = 2 * self.n
        self.neutral = (0, 0, False, False)
        self.st = [self.neutral] * size
        self.set_tag = [-1] * size

    def __operation(self, x, y):
        if x[3] and y[2]:
            return x[0] + y[0], x[1] + y[1] - 1, x[2], y[3]
        return x[0] + y[0], x[1] + y[1], x[2], y[3]

    def __push(self, node, l, r):
        if self.set_tag[node] != -1:
            v = self.set_tag[node]
            if v == 1:
                self.st[node] = (r - l + 1, 1, True, True)
            else:
                self.st[node] = (0, 0, False, False)
            if l != r:
                self.set_tag[node * 2] = v
                self.set_tag[node * 2 + 1] = v

            self.set_tag[node] = -1

    def range_set(self, ql, qr, v):
        self.__range_set(1, 0, self.n - 1, ql, qr, v)

    def __range_set(self, node, l, r, ql, qr, v):
        self.__push(node, l, r)
        if qr < l or r < ql:
            return
        if ql <= l and r <= qr:
            self.set_tag[node] = v
            self.__push(node, l, r)
        else:
            m = (l + r) // 2
            self.__range_set(node * 2, l, m, ql, qr, v)
            self.__range_set(node * 2 + 1, m + 1, r, ql, qr, v)
            self.st[node] = self.__operation(self.st[node * 2], self.st[node * 2 + 1])

    def query(self, ql, qr):
        return self.__query(1, 0, self.n - 1, ql, qr)[:2][::-1]

    def __query(self, node, l, r, ql, qr):
        self.__push(node, l, r)
        if qr < l or r < ql:
            return self.neutral
        if ql <= l and r <= qr:
            return self.st[node]
        m = (l + r) // 2
        return self.__operation(self.__query(node * 2, l, m, ql, qr), self.__query(node * 2 + 1, m + 1, r, ql, qr))


n = int(input())
a = [0] * (1 << 20)
ADDER = 500_000
st = MySegmentTree(a)

for _ in range(n):
    q = input().split()
    if q[0] == 'W':
        l = int(q[1]) + ADDER
        r = int(q[2]) + l - 1
        st.range_set(l, r, 0)
        print(' '.join(map(str, st.query(0, len(a)))))
    else:
        l = int(q[1]) + ADDER
        r = int(q[2]) + l - 1
        st.range_set(l, r, 1)
        print(' '.join(map(str, st.query(0, len(a)))))